python/lsst/pipe/tasks/htmIndexEngine.py

```diff
@@ -133,6 +133,135 @@ def run(self, parsedCmd):
                 result = result,
             )
 
+class IngestIndexedReferenceTask(pipeBase.CmdLineTask):
+    ConfigClass = IngestIndexedReferenceConfig
+    RunnerClass = IngestReferenceRunner
+    _DefaultName = 'IngestIndexedReferenceTask'
+
+    _flags = ['photometric', 'resolved', 'variable']
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        parser = pipeBase.ArgumentParser(name=cls._DefaultName)
+        parser.add_argument("files", nargs="+", help="Names of files to index")
+        return parser
+
+    def __init__(self, *args, **kwargs):
+        self.butler = kwargs.pop('butler')
+        pipeBase.Task.__init__(self, *args, **kwargs)
+        self.indexer = HtmIndexer(self.config.level)
+
+    def create_indexed_catalog(self, files):
+        rec_num = 0
+        first = True
+        for filename in files:
+            names = True
+            if self.config.colnames:
+                names = self.config.colnames
+            arr = numpy.genfromtxt(filename, dtype=None, skip_header=self.config.header_lines,
+                                   delimiter=self.config.delimiter,
+                                   names=names)
+            index_list = self.indexer.index_points(arr[self.config.ra_name], arr[self.config.dec_name])
+            if first:
+                schema = self.make_schema(arr.dtype)
+                #persist empty catalog to hold the master schema
+                dataId = self.make_data_id('master_schema')
+                self.butler.put(self.get_catalog(dataId, schema), self.config.ref_dataset_name,
+                                dataId=dataId)
+                first = False
+            pixel_ids = set(index_list)
+            shards = []
+            for pixel_id in pixel_ids:
+                dataId = self.make_data_id(pixel_id)
+                catalog = self.get_catalog(dataId, schema)
+                els = numpy.where(index_list == pixel_id)
+                #Just in case someone has only one line in the file.
+                for row in numpy.atleast_1d(arr)[els]:
```

Comment:
Can you add multiple rows at once (be sure to use `reserve` to allocate contiguous memory), and then set the columns for those rows?